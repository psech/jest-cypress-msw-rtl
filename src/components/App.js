import './App.css';
import Pizza from './Pizza';
import Toggle from './Toggle';
import Todos from './Todos';

function App() {
  return (
    <div className="App">
      <header className="App-header">
        <a
          className="App-link"
          href="https://reactjs.org"
          target="_blank"
          rel="noopener noreferrer"
        >
          Learn React!
        </a>
      </header>
      <Pizza ingredients={['bacon', 'ham']} />
      <Toggle />
      <Todos />
    </div>
  );
}

export default App;
