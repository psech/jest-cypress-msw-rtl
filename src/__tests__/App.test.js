import { render, screen } from '@testing-library/react';
import App from '../components/App';

test('renders learn react link', () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toBeInTheDocument();
});

test('verify a snapshot', () => {
  render(<App />);
  const linkElement = screen.getByText(/learn react/i);
  expect(linkElement).toMatchSnapshot();
});
