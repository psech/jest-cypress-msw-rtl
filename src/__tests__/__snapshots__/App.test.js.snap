// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`verify a snapshot 1`] = `
<a
  class="App-link"
  href="https://reactjs.org"
  rel="noopener noreferrer"
  target="_blank"
>
  Learn React!
</a>
`;
